Services/Registration/classes/class.ilAccountRegistrationGUI.php

```diff
@@ -349,24 +348,21 @@ public function saveForm()
         // validate username
         $login_obj = $this->form->getItemByPostVar('username');
         $login = $this->form->getInput("username");
+        $captcha = $this->form->getItemByPostVar("captcha_code");
         if (!ilUtil::isLogin($login)) {
             $login_obj->setAlert($this->lng->txt("login_invalid"));
             $form_valid = false;
-        } elseif (ilObjUser::_loginExists($login) && !empty($this->form->getInput("captcha_code"))) {
-            $login_obj->setAlert($this->lng->txt("login_exists"));
-            $form_valid = false;
-        } elseif(ilObjUser::_loginExists($login) && empty($this->form->getInput("captcha_code"))) {
+        } elseif (ilObjUser::_loginExists($login)) {
+            if(empty($captcha->getAlert())) {
+                $login_obj->setAlert($this->lng->txt("login_exists"));
```

Comment:
Have you tried to use $captcha->checkInput() ? 
I would prefer this query, in contrast to evaluating getAlert(). 
E. g see $captcha->checkInput() : i f $lng is not set alert is empty but checkout input failed.

Reply:
I did try that first, no matter the input however, it would always return false. That would only occur if the checkInput for captcha is called multiple times. So once when the whole form is validated and once when I specifically tried to check the input of the Captcha.

Therefore I decided to check for the alert, which is set when the captcha is checked the first time and stays that way.

I couldn't really find out why exactly captcha checkinput returns false when checked a second time, my best guess was that it probably checks the input against a different image.

Reply:
Ok, I understand. 
Thanks for fixing.